### PR TITLE
Generate the mocking framework (a bit)

### PR DESCRIPTION
**Pull request by DrDeano**
Instead of adding new types in 5 different places, this adds a generation tool that you add the type once and it will be populated in the framework for you.
This is defo one of those over engineered projects, but I had fun making it.

### CODE REVIEW

File: test/gen_types.zig [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+// All the function generation parts are the same apart from 3 things
+fn genGenericFunc(comptime intermediate: []const u8, comptime trail: []const u8, comptime end: []const u8) []const u8 {
+    comptime var str: []const u8 = "";
+    inline for (types) |t, i| {
+        if (i == 0) {
+            // Want nice formatting
+            str = str ++ "    " ++ t[0] ++ intermediate ++ t[1] ++ trail;
+        } else {
+            str = str ++ "                " ++ t[0] ++ intermediate ++ t[1] ++ trail;
+        }
```
Comment (SamTebbs33) [inline test/gen_types.zig]: We could probably remove the duplicate code by assigning the big ol' block of spaces to a const and using that.

**Review by SamTebbs33 — CHANGES_REQUESTED** [review]
This is indeed one of the most beautifully over-engineered solutions I have ever seen :D

File: test/gen_types.zig [hunk 2]
```diff
@@ -145,12 +147,12 @@ fn genGenericFunc(comptime intermediate: []const u8, comptime trail: []const u8,
     inline for (types) |t, i| {
         if (i == 0) {
             // Want nice formatting
-            str = str ++ "    " ++ t[0] ++ intermediate ++ t[1] ++ trail;
+            str = str ++ tab_space ++ t[0] ++ intermediate ++ t[1] ++ trail;
         } else {
-            str = str ++ "                " ++ t[0] ++ intermediate ++ t[1] ++ trail;
+            str = str ++ tab_space ** 4 ++ t[0] ++ intermediate ++ t[1] ++ trail;
         }
     }
```
Comment (SamTebbs33) [inline test/gen_types.zig]: I meant something a bit more like
```
const spaces = if (i == 0) " " ** 4 else " " ** 16;
str = str ++ spaces ++ t[0] ++ intermediate ++ t[1] ++ trail;
```

**Review by SamTebbs33 — CHANGES_REQUESTED** [review]

File: test/gen_types.zig (hunk 2 above)
Comment (DrDeano) [reply test/gen_types.zig]: Sure

**Review by SamTebbs33 — APPROVED** [review]
lgtm

**Pull request merged**